### PR TITLE
Add transaction counting capability

### PR DESCRIPTION
**Pull request by ashanhr**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/TransactionCountDAO.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
+package org.wso2.carbon.apimgt.impl.dao;
```
Comment (RakhithaRR) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/TransactionCountDAO.java:1]: Add license texts to all new files.

Comment (ashanhr) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/TransactionCountDAO.java:18]: Resolved

**Review by AnuGayan — APPROVED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/TransactionCountDAO.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+package org.wso2.carbon.apimgt.impl.dao;
+
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.api.ExceptionCodes;
+import org.wso2.carbon.apimgt.impl.dao.constants.SQLConstants;
+import org.wso2.carbon.apimgt.impl.dto.TransactionCountDTO;
+import org.wso2.carbon.apimgt.impl.utils.APIMgtDBUtil;
+
+import java.sql.*;
```
Comment (npamudika) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/TransactionCountDAO.java:27]: Let's avoid wild card imports

File: components/apimgt/org.wso2.carbon.apimgt.internal.service/src/main/java/org/wso2/carbon/apimgt/internal/service/impl/TransactionRecordsApiServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+package org.wso2.carbon.apimgt.internal.service.impl;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.impl.dao.TransactionCountDAO;
+import org.wso2.carbon.apimgt.impl.dto.TransactionCountDTO;
+import org.wso2.carbon.apimgt.internal.service.*;
+import org.wso2.carbon.apimgt.internal.service.dto.*;
```
Comment (npamudika) [inline components/apimgt/org.wso2.carbon.apimgt.internal.service/src/main/java/org/wso2/carbon/apimgt/internal/service/impl/TransactionRecordsApiServiceImpl.java:27]: Let's avoid wild card imports. Please correct in other places too.

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SQLConstants.java [hunk 4]
```diff
@@ -4408,4 +4408,12 @@ public static class SystemConfigsConstants {
                 + "SET CONFIGURATION = ? WHERE ORGANIZATION = ? AND CONFIG_TYPE = ?";
     }
 
+    /**
+     * Static class to hold database queries related to AM_TRANSACTION_RECORDS table
+     */
+    public static class TransactionCountConstants {
+        public static final String INSERT_TRANSACTION_COUNT = "INSERT INTO AM_TRANSACTION_RECORDS " + "(ID, HOST, SERVER_ID, SERVER_TYPE, COUNT, RECORDED_TIME) " + "VALUES (?,?,?,?,?,?)";
```
Comment (ashanhr) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/constants/SQLConstants.java:4415]: Line length is fixed by https://github.com/wso2/carbon-apimgt/commit/54ae42a616323a1d01d41d2af78e1a824371b67f

**Pull request merged**